core/src/main/scala/com/rallyhealth/vapors/factfilter/data/Fact.scala

```diff
@@ -20,6 +22,36 @@ object Fact {
     factType: FactType[T],
     value: T,
   ): Fact = TypedFact(factType, value)
+
+  val orderByFactValue: Order[Fact] = { (x, y) =>
+    def maybeXCompared =
+      x.typeInfo
+        .cast(y)
+        .map(yAsX => x.typeInfo.order.compare(x.value, yAsX.value))
+
+    def maybeYCompared =
+      y.typeInfo
+        .cast(x)
+        .map(xAsY => y.typeInfo.order.compare(xAsY.value, y.value))
+
+    def fallbackToCompareAsStrings =
+      if (x.value == y.value) 0
+      else Order[String].compare(x.value.toString, y.value.toString)
+
+    maybeXCompared.orElse(maybeYCompared).getOrElse(fallbackToCompareAsStrings)
+  }
+
+  def orderByFactName(nameOrder: Order[String]): Order[Fact] = {
+    nameOrder.contramap(_.typeInfo.name)
+  }
+
+  // This will be used a lot, so cache it
+  private final val defaultOrder: Order[Fact] = orderByFactName(Order[String])
+
+  implicit def order(implicit orderFactNames: Order[String]): Order[Fact] = {
```

Comment:
So, these are mostly compared as `String`s?

Reply:
yea, it is mostly a wrapper around a String with some type information.